### PR TITLE
LUCENE-9394: fix and suppress warnings

### PR DESCRIPTION
**Pull request by msokolov**

### CODE REVIEW

File: lucene/core/src/test/org/apache/lucene/analysis/TestCharArraySet.java [hunk 1]
```diff
@@ -61,15 +61,17 @@ public void testNonZeroOffset() {
   public void testObjectContains() {
     CharArraySet set = new CharArraySet(10, true);
     Integer val = Integer.valueOf(1);
+    @SuppressWarnings("deprecation")
+    Integer val1 = new Integer(1);
```
Comment (madrob) [inline lucene/core/src/test/org/apache/lucene/analysis/TestCharArraySet.java:65]: Add a comment that we're explicitly avoiding the Integer cache, and an `assertNotSame(val, val1)`?

File: lucene/spatial-extras/src/test/org/apache/lucene/spatial/prefix/HeatmapFacetCounterTest.java [hunk 2]
```diff
@@ -33,11 +33,7 @@
 import org.locationtech.spatial4j.context.SpatialContext;
 import org.locationtech.spatial4j.context.SpatialContextFactory;
 import org.locationtech.spatial4j.distance.DistanceUtils;
-import org.locationtech.spatial4j.shape.Circle;
-import org.locationtech.spatial4j.shape.Point;
-import org.locationtech.spatial4j.shape.Rectangle;
-import org.locationtech.spatial4j.shape.Shape;
-import org.locationtech.spatial4j.shape.SpatialRelation;
+import org.locationtech.spatial4j.shape.*;
```
Comment (madrob) [inline lucene/spatial-extras/src/test/org/apache/lucene/spatial/prefix/HeatmapFacetCounterTest.java:36]: wildcard import

File: lucene/spatial-extras/src/test/org/apache/lucene/spatial/spatial4j/RandomizedShapeTestCase.java [hunk 3]
```diff
@@ -183,106 +179,4 @@ private void _assertIntersect(String msg, SpatialRelation expected, Shape a, Sha
     }
   }
 
-  protected void assertEqualsRatio(String msg, double expected, double actual) {
```
Comment (madrob) [inline lucene/spatial-extras/src/test/org/apache/lucene/spatial/spatial4j/RandomizedShapeTestCase.java:186]: There appear to be more unused methods in this class, why did we keep them but not these?

**Comment by msokolov** [comment]
Thanks for the comments, @madrob, I posted a new PR addressing them. I'm not sure how I missed all that unused code in RandomizedShapeTestCase - it's pretty bare now!

**Comment by ErickErickson** [comment]
speaking from experience, when dealing with this many changes in a big wodge, it’s _very easy_ to have some things slip through ;)

**Pull request merged**